pkg/azurefile/controllerserver.go

```diff
@@ -1050,31 +1081,69 @@ func (d *Driver) getServiceURL(ctx context.Context, sourceVolumeID string, secre
-	if err != nil {
-		return false, azfile.ShareItem{}, err
-	}
-	for _, share := range listSnapshot.ShareItems {
-		if share.Metadata[snapshotNameKey] == snapshotName {
-			if share.Name == fileShareName {
-				klog.V(2).Infof("found share(%s) snapshot(%s) Metadata(%v)", share.Name, *share.Snapshot, share.Metadata)
-				return true, share, nil
+		// List share snapshots.
+		listSnapshot, err := serviceURL.ListSharesSegment(ctx, azfile.Marker{}, azfile.ListSharesOptions{Detail: azfile.ListSharesDetail{Metadata: true, Snapshots: true}})
+		if err != nil {
+			return false, "", time.Time{}, 0, err
+		}
+		for _, share := range listSnapshot.ShareItems {
+			if share.Metadata[snapshotNameKey] == snapshotName {
+				if share.Name == fileShareName {
+					klog.V(2).Infof("found share(%s) snapshot(%s) Metadata(%v)", share.Name, *share.Snapshot, share.Metadata)
+					if share.Snapshot == nil {
+						return true, "", share.Properties.LastModified, share.Properties.Quota, status.Errorf(codes.Internal, "Snapshot property of %s is nil", share.Name)
+					}
+					return true, *share.Snapshot, share.Properties.LastModified, share.Properties.Quota, nil
+				}
+				return true, "", time.Time{}, 0, fmt.Errorf("snapshot(%s) already exists, while the current file share name(%s) does not equal to %s, SourceVolumeId(%s)", snapshotName, share.Name, fileShareName, sourceVolumeID)
 			}
-			return true, azfile.ShareItem{}, fmt.Errorf("snapshot(%s) already exists, while the current file share name(%s) does not equal to %s, SourceVolumeId(%s)", snapshotName, share.Name, fileShareName, sourceVolumeID)
 		}
-	}
+	} else {
+		rgName, accountName, fileShareName, _, _, subsID, err := GetFileShareInfo(sourceVolumeID) //nolint:dogsled
+		if err != nil {
+			return false, "", time.Time{}, 0, err
+		}
+		if fileShareName == "" {
+			return false, "", time.Time{}, 0, fmt.Errorf("file share is empty after parsing sourceVolumeID: %s", sourceVolumeID)
+		}
 
-	return false, azfile.ShareItem{}, nil
+		// List share snapshots.
+		listSnapshot, err := d.cloud.FileClient.WithSubscriptionID(subsID).ListFileShare(ctx, rgName, accountName, "", "snapshots")
+		if err != nil {
+			return false, "", time.Time{}, 0, err
+		}
+		if listSnapshot == nil {
+			return false, "", time.Time{}, 0, nil
+		}
+		for _, share := range listSnapshot {
+			if share.SnapshotTime == nil { //the fileshare is not a snapshot
+				continue
+			}
+			fileshare, err := d.cloud.FileClient.WithSubscriptionID(subsID).GetFileShare(ctx, rgName, accountName, *share.Name, share.SnapshotTime.Format(snapshotTimeFormat))
+			if err != nil {
+				klog.V(2).Infof("get share(%s) snapshot(%s) error(%s)", *share.Name, share.SnapshotTime.Format(snapshotTimeFormat), err)
+				return false, "", time.Time{}, 0, nil
+			}
+			if fileshare.Metadata != nil && *fileshare.Metadata[snapshotNameKey] == snapshotName {
+				if *fileshare.Name == fileShareName {
```

Comment:
use `to.String(fileshare.Name)` to get string, and if `fileshare.Name` is nil, it won't panic